TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/subsystems/XDrive.java

```diff
@@ -0,0 +1,187 @@
+        back_left = hardwareMap.dcMotor.get(MOTOR_BACK_LEFT);
+        back_right = hardwareMap.dcMotor.get(MOTOR_BACK_RIGHT);
+        front_right = hardwareMap.dcMotor.get(MOTOR_FRONT_RIGHT);
+
+        front_left.setDirection(DcMotor.Direction.REVERSE);
+        back_left.setDirection(DcMotor.Direction.REVERSE);
+        front_right.setDirection(DcMotor.Direction.FORWARD);
+        back_right.setDirection(DcMotor.Direction.FORWARD);
+
+        front_left.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+        back_left.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+        front_right.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+        back_right.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+
+        this.telemetry = telemetry;
+
+        imu = hardwareMap.get(IMU.class, "imu");
+        IMU.Parameters parameters = new IMU.Parameters(
+                new RevHubOrientationOnRobot(
+                        RevHubOrientationOnRobot.LogoFacingDirection.DOWN,
+                        RevHubOrientationOnRobot.UsbFacingDirection.FORWARD
+                )
+        );
+        imu.initialize(parameters);
+    }
+
+    @Override
+    public void start() {
+    }
+
+    @Override
+    public void stop() {
+    }
+
+    @Override
+    public void execute(GamepadManager gamepadManager) {
+        drive(gamepadManager.getDriver());
+        resetAngle(gamepadManager.getDriver());
+    }
+
+    public void drive(SmartGamepad driver) {
+        double rotate = driver.getRightStickX()/4;
+        double stick_x = driver.getLeftStickX() * Math.sqrt(Math.pow(1-Math.abs(rotate), 2)/2);
+        double stick_y = driver.getLeftStickY() * Math.sqrt(Math.pow(1-Math.abs(rotate), 2)/2);
+        double theta = 0;
+        double Px = 0;
+        double Py = 0;
+
+        double gyroAngle = getHeading() * Math.PI / 180;
+        if (gyroAngle <= 0) {
+            gyroAngle = gyroAngle + (Math.PI / 2);
+        } else if (0 < gyroAngle && gyroAngle < Math.PI / 2) {
+            gyroAngle = gyroAngle + (Math.PI / 2);
+        } else if (Math.PI / 2 <= gyroAngle) {
+            gyroAngle = gyroAngle - (3 * Math.PI / 2);
+        }
+        gyroAngle = -1 * gyroAngle;
+
+        if(driver.isButtonRightBumper()){
+            gyroAngle = -Math.PI/2;
+        }
+
+        if(driver.isButtonDPadRight()){
+            stick_x = 0.5;
+        }
+        else if(driver.isButtonDPadLeft()){
+            stick_x = -0.5;
+        }
+        if(driver.isButtonDPadUp()){
+            stick_y = -0.5;
+        }
+        else if(driver.isButtonDPadDown()){
+            stick_y = 0.5;
+        }
+
+
+        //MOVEMENT
+        theta = Math.atan2(stick_y, stick_x) - gyroAngle - (Math.PI / 2);
+        Px = Math.sqrt(Math.pow(stick_x, 2) + Math.pow(stick_y, 2)) * (Math.sin(theta + Math.PI / 4));
+        Py = Math.sqrt(Math.pow(stick_x, 2) + Math.pow(stick_y, 2)) * (Math.sin(theta - Math.PI / 4));
+
+        telemetry.addData("Stick_X", stick_x);
```

Comment:
as a suggestion to the lib as well, like frc wpi has a space for logging, add to the interface a method where you should add all default logs

Reply:
I was in a call with a ex member of an frc team and we talked about a logging system. I also suggested the Lib for him and said that we would update the lib and that we were thinking of making a logging system. On the offseason it's a good ideia to make one